snow/engine/snowman/bootstrap/bootstrapper.go

```diff
@@ -603,25 +617,36 @@ func (b *Bootstrapper) process(
 		height := blk.Height()
 		b.tipHeight = max(b.tipHeight, height)
 
-		if numPreviouslyFetched/statusUpdateFrequency != numFetched/statusUpdateFrequency {
+		// Check if it's time to log progress (both progress-based and time-based frequency)
+		now := time.Now()
+		shouldLog := numPreviouslyFetched/statusUpdateFrequency != numFetched/statusUpdateFrequency &&
+			now.Sub(b.lastProgressUpdateTime) >= minimumLogInterval
+
+		if shouldLog {
 			totalBlocksToFetch := b.tipHeight - b.startingHeight
-			eta := timer.EstimateETA(
-				b.startTime,
-				numFetched-b.initiallyFetched,         // Number of blocks we have fetched during this run
-				totalBlocksToFetch-b.initiallyFetched, // Number of blocks we expect to fetch during this run
+
+			eta, progressPercentage := b.etaTracker.AddSample(
```

Comment:
to align with the other callsite
```suggestion
			etaPtr, progressPercentage := b.etaTracker.AddSample(
```